src/compiler/c.rs

```diff
@@ -245,8 +245,26 @@ impl<T, I> CompilerHasher<T> for CCompilerHasher<I>
         });
         let out_pretty = parsed_args.output_pretty().into_owned();
         let extra_hashes = hash_all(&parsed_args.extra_hash_files, &pool.clone());
+        let outputs = parsed_args.outputs.clone();
+        let args_cwd = cwd.clone();
 
         Box::new(result.or_else(move |err| {
+            // Errors remove all traces of potential output.
+            debug!("removing files {:?}", &outputs);
+
+            let v: std::result::Result<Vec<()>, std::io::Error> = outputs.values().map(|f| {
```

Comment:
The Vec of empty values we don't consult seems unnecessary, maybe this can be something like a `fold`?

Reply:
The motivation was not worrying too much about the performance of error cases, but I like the `fold` idea better.  I'll use that, thanks!